packages/core/src/event_emitter.ts

```diff
@@ -163,6 +141,17 @@ class EventEmitter_ extends Subject<any> {
 
     return sink;
   }
+
+  private createHanlder(callback: (...args: any) => void) {
+    if (this.__isAsync) {
+      return (...args: any) => {
+        setTimeout(() => callback(...args));
+      };
+    }
+    return (...args: any) => {
+      callback(...args);
+    };
+  }
```

Comment:
Can we just return callback function like this?
```suggestion
    return callback;
  }
```

Reply:
Yeah, looks good